esp-hal/src/lcd_cam/lcd/i8080.rs

```diff
@@ -478,15 +493,15 @@ impl<'d, BUF, CH: DmaChannel> I8080Transfer<'d, BUF, CH, crate::Async> {
     }
 }
 
-impl<'d, BUF, CH: DmaChannel, DM: Mode> Drop for I8080Transfer<'d, BUF, CH, DM> {
+impl<'d, BUF: DmaTxBuffer, CH: DmaChannel, DM: Mode> Drop for I8080Transfer<'d, BUF, CH, DM> {
     fn drop(&mut self) {
         self.stop_peripherals();
 
-        // SAFETY: This is Drop, we know that self.i8080 and self.tx_buf
+        // SAFETY: This is Drop, we know that self.i8080 and self.buf_view
         // won't be touched again.
         unsafe {
             ManuallyDrop::drop(&mut self.i8080);
-            ManuallyDrop::drop(&mut self.tx_buf);
+            ManuallyDrop::drop(&mut self.buf_view);
```

Comment:
IMHO it would be more correct to reconstruct the buffer first and drop that. There are no restrictions on `DmaTRxBuffer::View` but the trait itself is unsafe so we may prescribe expected Drop behaviour there.

Reply:
Agreed, will update!